tests/subsys/pm/power_mgmt/src/dummy_driver.c

```diff
@@ -9,33 +9,88 @@
 #include <pm/device_runtime.h>
 #include "dummy_driver.h"
 
+static bool dont_sleep;
+extern bool z_sys_post_kernel;
+
 static int dummy_open(const struct device *dev)
 {
-	return pm_device_get(dev);
+	dont_sleep = false;
+	pm_device_get(dev);
+	if (dev->pm->enable == false) {
+		pm_device_enable(dev);
+		return pm_device_get(dev);
+
+	}
+	return 0;
 }
 
 static int dummy_close(const struct device *dev)
 {
-	return pm_device_put(dev);
+	dont_sleep = false;
+	pm_device_put(dev);
+	if (z_sys_post_kernel == false) {
+		z_sys_post_kernel = true;
+		return 0;
+	}
+	pm_device_disable(dev);
+	return 0;
+}
+
+static int dummy_refuse_to_sleep(const struct device *dev)
+{
+	dont_sleep = true;
+	return 0;
+}
+
+static int dummy_not_wakeup_capable(const struct device *dev)
+{
+	atomic_clear_bit(&dev->pm->flags, PM_DEVICE_FLAGS_WS_CAPABLE);
+	return 0;
+}
+
+static int dummy_get_ready_to_close(const struct device *dev)
+{
+	dev->pm->state = PM_DEVICE_STATE_SUSPENDING;
+	return 0;
+}
+
+static int dummy_sys_pre_kernel(const struct device *dev)
+{
+	z_sys_post_kernel = false;
```

Comment:
you should not manipulate this variable directly IMHO.

Reply:
yes, updated